pkg/config/expand.go

```diff
@@ -218,12 +222,27 @@ func (dc *DeploymentConfig) applyUseModules() error {
 			fromModInputs := getModuleInputMap(fromModInfo.Inputs)
 			settingsInBlueprint := maps.Keys(fromMod.Settings)
 			for _, toModID := range fromMod.Use {
-				toMod, err := group.getModuleByID(toModID)
+				modRef, err := identifyModuleByReference(toModID, *group)
```

Comment:
I will improve the commenting here.

Reply:
Nit: PS: I think commenting in its final form is is too verbose. Will share reference separately. I was more thinking of some white space and a single line comment in this specific location. It is probably a sign that the code has become two dense and needs some refactoring or breaking out into separate functions.

Reply:
I agree that the verbosity is a sign that this area of the code needs a rethink. I aim to document lessons here with (at least) toy examples of how it could be improved conceptually and technically. The big problem here is that the values stay "raw" (e.g. strings) even after they have been validated.